master/buildbot/test/integration/test_graphql_queries.yaml

```diff
@@ -0,0 +1,144 @@
+- query: |
+    {masters{name, builders(name:"slowruntests"){name, builds(limit:1){number}}}}
+  xpected:
+    masters:
+    - builders:
+      - builds:
+        - number: 1
+        name: slowruntests
+      name: master1:/workdir
+- query: |
+    {builders(name:"slowruntests"){name, builds(limit:1){number}}}
+  xpected:
```

Comment:
I think `xpected` is not a big problem, just that it's weird to see a typo :-) I agree with reasoning about sorting. If switching to ruaml is not too hard I think it would make sense to do that.